rfcs/text/0008-threat-intel.md

```diff
@@ -75,7 +81,178 @@ Stage 3: Add or update all remaining field definitions. The list should now be e
+       e.g. sha256, sha1, ssdeep, etc. If that's the case this should be an array
+       of types (i.e. [sha1, sha256, ssdeep]) */
+    "ioc.type": ["sha256", "md5", "file_name", "file_size"],
+    "ioc.description": "file last associated with delivering Angler EK",
+
+    // These would provide attribution-related information for the IOCs
+    "malware": {
+        "name": "CryptoMinerX.B",
+        "family": "CryptoMinerX",
+        "type": "cryptominer"
+    },
+    "threat_actor": {
+        "name": "CryptoCurrency R Us",
+        "type": [
+            "criminal"
+        ]
+    },
+
+    // Filebeats and other fields, not part of ECS proposal
+    "fileset.name": "abusemalware",
+    "input.type": "log",
+    "log.offset": 0,
+
+    // Any indicators should also be copied to relevant related.* field
+    "related": {
+        "hash": [
+            "1eee2bf3f56d8abed72da2bc523e7431",
+            "0c415dd718e3b3728707d579cf8214f54c2942e964975a5f925e0b82fea644b4"
+        ]
+    },
+    "tags": [
+        "threatintel",
+        "forwarded"
+    ],
+}
+```
+
+2. **Adding threat intelligence match/enrichment to another document which could be in a source event index or signals index.** 
+
+    The Indicator Match Rule will be used to generate signals when a match occurs between a source event and threat intelligence document. The ECS fields proposed here will be used to add the enrichment and threat intel context in the signal document. 
+
+**Example**
+```json5
+{
+    "process": {
+        "name": "svchost.exe",
+        "pid": 1644,
+        "entity_id": "MDgyOWFiYTYtMzRkYi1kZTM2LTFkNDItMzBlYWM3NDVlOTgwLTE2NDQtMTMyNDk3MTA2OTcuNDc1OTExNTAw",
+        "executable": "C:\\Windows\\System32\\svchost.exe"
+    },
+    "message": "Endpoint file event",
+    "@timestamp": "2020-11-17T19:07:46.0956672Z",
+    "file": {
+        "path": "C:\\Windows\\Prefetch\\SVCHOST.EXE-AE7DB802.pf",
+        "extension": "pf",
+        "name": "SVCHOST.EXE-AE7DB802.pf",
+        "hash": {
+            "sha256": "0c415dd718e3b3728707d579cf8214f54c2942e964975a5f925e0b82fea644b4"
+        }
+    },
+    "threat": [
+        // Each enrichment is added as a nested object under `threat.*`
+        {
+            "ioc": {
+                // Copy all the object IOCs under `ioc.*`, providing full context
+                "file": {
+                    "hash": {
+                        "sha256": "0c415dd718e3b3728707d579cf8214f54c2942e964975a5f925e0b82fea644b4",
+                        "md5": "1eee2bf3f56d8abed72da2bc523e7431"
+                    },
+                    "size": 656896,
+                    "name": "invoice.doc"
+                },
+                /* `matched` will provide context about which of the IOCs above matched on this
+                   particular enrichment. If multiple matches for this IOC object, this could
+                   be a list */
+                "matched": "sha256",
+                "marking": {
```

Comment:
Here, `marking` resides at `threat.ioc.marking.*`, but the field listing reflects `threat.marking.*`. Expected?

Reply:
this is in the threat intel index, but the marking applies to the IOC. We propose the ECS fields as stored in the enriched document, but the structure for the threat index needs to be a bit different to facilitate better analysis across all documents, including those that are not enriched (maybe the intel match arrived after the document was ingested, for example).